### PR TITLE
Upgrade to Groovy 3.0.7

### PR DESCRIPTION
**Pull request by dreis2211**
Hi,

this PR tackles #20119 as discussed.

Let me know what you think.
Cheers,
Christoph

### CODE REVIEW

File: spring-boot-project/spring-boot-cli/src/test/java/org/springframework/boot/cli/compiler/ResolveDependencyCoordinatesTransformationTests.java [hunk 1]
```diff
@@ -87,7 +87,8 @@ void setUpExpectations() {
 
 	@Test
 	void transformationOfAnnotationOnImport() {
-		this.moduleNode.addImport(null, null, Arrays.asList(this.grabAnnotation));
+		ClassNode classNode = new ClassNode("Test", 0, new ClassNode(Object.class));
```
Comment (dreis2211) [inline spring-boot-project/spring-boot-cli/src/test/java/org/springframework/boot/cli/compiler/ResolveDependencyCoordinatesTransformationTests.java:90]: Passing `null` throws an NPE these days.

**Comment by dreis2211** [comment]
Apparently, I need to check why my local builds didn't report the same build failure as on Concourse. I will work on the build error of course.

File: spring-boot-project/spring-boot-cli/samples/http.groovy [hunk 2]
```diff
@@ -17,7 +16,7 @@ class Example implements CommandLineRunner {
 
 	void run(String... args) {
 		def port = context.webServer.port
-		def world = new RESTClient("http://localhost:" + port).get(path:"/").data.text
+		def world = new RestTemplate().getForObject("http://localhost:" + port + "/", String.class);
```
Comment (szpak) [inline spring-boot-project/spring-boot-cli/samples/http.groovy:19]:
```suggestion
		def world = new RestTemplate().getForObject("http://localhost:$port/", String.class);
```

It's more the Groovy way :).

Comment (szpak) [reply spring-boot-project/spring-boot-cli/samples/http.groovy:19]: Anyway, I would put here `int` and `String` instead of just `def` to clarity.

**Review by szpak — APPROVED** [review]

**Review by szpak — APPROVED** [review]
In general LGTM.

File: spring-boot-project/spring-boot-cli/samples/http.groovy (hunk 2 above)
Comment (wilkinsona) [reply spring-boot-project/spring-boot-cli/samples/http.groovy:19]: Thanks for taking a look, @szpak. Those are good changes to make, but they're unrelated to an upgrade to Groovy 3.0 so let's keep them separate from this PR please.

**Comment by wilkinsona** [comment]
Thank you very much, @dreis2211. Much appreciated.

**Pull request closed without merging**